django/contrib/auth/password_validation.py

```diff
@@ -170,14 +170,22 @@ class CommonPasswordValidator:
     DEFAULT_PASSWORD_LIST_PATH = Path(__file__).resolve().parent / 'common-passwords.txt.gz'
 
     def __init__(self, password_list_path=DEFAULT_PASSWORD_LIST_PATH):
+        # Be careful about de-duplicating the set comprehension with
+        # a try/except/with combination; OSError will only be raised
+        # after iterating through the file, *not* directly from
+        # gzip.open().
         try:
-            with gzip.open(str(password_list_path)) as f:
-                common_passwords_lines = f.read().decode().splitlines()
+            # Pass encoding explicitly because common-passwords.txt.gz
+            # is distributed with the Django source.
```

Comment:
I think this change should be undone. Python 3 (normally) defaults to `utf-8`... and when it doesn't, this change introduces the chance of a regression. The default file isn't the only possible gzipped file.

Reply:
Yes, I see what you're saying here - you want to be careful of a situation where user passes a custom (gzip) `password_list_path` that is _not_ UTF-8 encoded.

What I was trying to do here was be explicit about UTF-8 but only in the narrow case for `DEFAULT_PASSWORD_LIST_PATH` (i.e., force that file to be unzipped with utf-8 decode even if the user's default encoding is different).  But I think that's not needed and both sides can be happy:

```python
In [1]: import gzip                                                                                                                                          

In [2]: from django.contrib.auth.password_validation import CommonPasswordValidator                                                                          

In [3]: with gzip.open(str(CommonPasswordValidator.DEFAULT_PASSWORD_LIST_PATH), "rt") as f: 
   ...:     pw = {x.strip() for x in f} 
   ...:                                                                                                                                                      

In [4]: all(i.isascii() for i in pw)                                                                                                                         
Out[4]: True
```